### PR TITLE
docs: add GLOSSARY.md

### PR DESCRIPTION
**Pull request by felangel**
## Description

- docs: add GLOSSARY.md

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [X] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: GLOSSARY.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+## Glossary
+
+Before getting started, we recommend familiarizing yourslef with the following terms and their definitions within the Shorebird ecosystem.
```
Comment (eseidel) [inline GLOSSARY.md]: yourself

Comment (eseidel) [reply GLOSSARY.md]: Although, honestly I'd remove this.  I'm not sure folks read these things first. 🤣  Some do I'm sure. 🤷

File: GLOSSARY.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+### Patch
+
+**Definition**
+
+An over the air update which is applied to a specific release.
+
+**Example**
+
+If we had an existing `1.0.0` release of `time_shift` available, we could publish a patch to `version 1.0.0`. All patches have a patch number (auto-incrementing integer) and multiple patches can be published for a given app release version.
```
Comment (eseidel) [inline GLOSSARY.md]: ... that would result in "patch 1", there can be many patches to a given release, typically the latest is active and applied to devices in the field.

File: GLOSSARY.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+**Example**
+
+We can create a "development" and "staging" channels in addition to the "stable" channel and promote patches to those internal channels for testing before rolling them out to the general public.
+
+### Artifact
+
+**Definition**
+
+An artifact contains metadata about the contents of a specific patch for a specific platform and architecture.
```
Comment (eseidel) [inline GLOSSARY.md:47]: Oh, the Artifact is the blob + the metadata imo.

**Review by eseidel — APPROVED** [review]

File: GLOSSARY.md (hunk 3 above)
Comment (felangel) [reply GLOSSARY.md:47]: I considered the download_url for the blob part of the artifact's metadata. Is that what you had in mind as well or did you imagine an artifact containing the blob itself? Maybe we need a different name for this 🤷

File: GLOSSARY.md (hunk 1 above)
Comment (felangel) [reply GLOSSARY.md]: Yeah good catch just removed it for now as you suggested

File: GLOSSARY.md (hunk 3 above)
Comment (felangel) [reply GLOSSARY.md:47]: Will merge this for now so we have a starting point but we can adjust this as needed in subsequent PRs 👍

**Pull request merged**